### PR TITLE
Version bumping 

### PR DESCRIPTION
**Pull request by jonoabroad**
Sorry, this could have been several PRs, and on the edge of opinionated, possibly wildly incorrectly :)

1. I've updated the open telemetry .net dependencies as there are some vulnerabilities,
2. I've updated the containers to the latest versions - just because they are shiny 
3. I've added the open telemetry collector grafana demo dashboard
4.  I've updated the open telemetry collector to including batching ( so the dashboard above looks a bit better), include it's open metrics and allow prometheus to pull metrics. 

Anyway, let me know if this is heading away from what you were thinking.

### CODE REVIEW

**Comment by Depechie** [comment]
@jonoabroad all great additions!
But somehow we now lost the info in the Aspire dashboard for traces, I tried reverting some parts without luck. I thought it would only be because you removed the OTEL.yml config for it, but that did not fix it.

I did not check this in a while so I need to first asses that and come back to the PR.
So, maybe, are you willing to do actual separate PR's? That way I can approve some already...

**Comment by jonoabroad** [comment]
I can have a look - I wasn't sure that was ever working for me, which is
why I got rid of it.

Let me go back to your original version and step forward from there.

Cheers for the quick response.

**Comment by Depechie** [comment]
Extra side note... maybe you did not notice it yet, but somehow if you only do PR's on branches you are not listed on the main page of the github repo.
Maybe if you like you can do a small PR on https://github.com/Depechie/OpenTelemetryGrafana the main branch to ( just bump the docker images versions or something easy ) so you are also listed there?

**Comment by jonoabroad** [comment]
Will have a look.

I can't get your original branch running, I remember now that post a
windows and docker desktop update aspire preview 4 refuses to start on my
machine.

Were you seeing the the traces in the aspire dashboard as well as grafana?

**Comment by Depechie** [comment]
Yes indeed I was seeing the trace info both in the Aspire dashboard as in Grafana. ( I now have some time to look at it again so will also try to see when I stopped getting them )
Do mind the main branch does not use Aspire.

**Comment by jonoabroad** [comment]
see comments interleaved

On Tue, 16 Apr 2024 at 06:49, Glenn Versweyveld ***@***.***>
wrote:

> Yes indeed I was seeing the trace info both in the Aspire dashboard as in
> Grafana. ( I now have some time to look at it again so will also try to see
> when I stopped getting them )
>
Thank you! I did want this, but put it down as a nice to have.


> Do mind the main branch does not use Aspire.
>
I don't mind, but also it is your house so I'm just trying not to move the
furniture around too much for my own benefit.

**Comment by Depechie** [comment]
Ok no clue why the export from OTEL collector to the Aspire dashboard is no longer working... even if I just set the URL fro; the LaunchSettings variable DOTNET_DASHBOARD_OTLP_ENDPOINT_URL straight in the otel.yml config, I get `connection refused` in the logs of the otel collector to that endpoint.

Also reverting to prev4 or prev3 is a no go somehow, if I do that, the app.host will not start and give a timeout error after 1 minute.

**Comment by jonoabroad** [comment]
Okay, this is the same error I was getting, so it's not just me.

Before logging an issue, I am going to have a play with the standalone
dashboard

File: code/config/otel.yml [hunk 1]
```diff
@@ -2,14 +2,20 @@ receivers:
...
+  prometheus:
+    trim_metric_suffixes: true
+    use_start_time_metric: true
+    start_time_metric_regex: .*
+    config:
+      scrape_configs:
+        - job_name: 'otel-collector-metrics'
+          scrape_interval: 5s
+          static_configs:
+            - targets: ['127.0.0.1:8888']
```
Comment (Depechie) [inline code/config/otel.yml:14]: Would it be possible to also fill these addresses in through environment vars?

File: code/config/otel.yml [hunk 2]
```diff
@@ -18,12 +24,24 @@ exporters:
     endpoint: ${env:LOKI_URL}
     tls:
       insecure: true
+  prometheus:
+    endpoint: 0.0.0.0:8889
```
Comment (Depechie) [inline code/config/otel.yml:28]: Would it be possible to also fill these addresses in through environment vars?

**Review by Depechie — COMMENTED** [review]
Left some comments

File: code/config/otel.yml [hunk 3]
```diff
@@ -18,12 +24,23 @@ exporters:
...
+
+processors:
+  batch:
 
 service:
   pipelines:
     logs:
       receivers: [otlp]
-      exporters: [loki, otlp/dashboard, debug]
```
Comment (Depechie) [inline code/config/otel.yml:26]: This needs to be added back if we also want the Aspire dashboard to represent the otel information.

File: code/config/otel.yml [hunk 4]
```diff
@@ -18,12 +24,23 @@ exporters:
...
+      processors: [batch]
+      exporters: [loki,debug]
+    metrics:
+      receivers: [prometheus,otlp]
+      processors: [batch]
+      exporters: [prometheus,debug]
     traces:
       receivers: [otlp]
-      exporters: [otlp, otlp/dashboard, debug]
```
Comment (Depechie) [inline code/config/otel.yml:29]: This needs to be added back if we also want the Aspire dashboard to represent the otel information.

File: code/config/otel.yml [hunk 5]
```diff
@@ -2,14 +2,20 @@ receivers:
...
+      scrape_configs:
+        - job_name: 'otel-collector-metrics'
+          scrape_interval: 5s
+          static_configs:
+            - targets: ['127.0.0.1:8888']
 
 exporters:
   debug:
     verbosity: detailed
-  otlp/dashboard:
```
Comment (Depechie) [inline code/config/otel.yml:9]: This needs to be added back if we also want the Aspire dashboard to represent the otel information.

**Review by Depechie — COMMENTED** [review]
Added extra comments

**Pull request merged**